### DISCUSSION TITLE
What sort of error can i expect when running out of gas using singlepass backend?

### DISCUSSION THREAD

**Discussion by reuvenpo** [wasmerio/wasmer#1452] [Q&A]
I'm playing around with a system that uses the singlepass backend (v0.17) and runs some wasm module (the details here don't matter, really). I can see that when providing it with enough gas, it completes successfully, but when I give it even one gas unit below some threshold, the wasmer instance returns `RuntimeError::InvokeError(InvokeError::FailedWithNoError)`. I tried this with various gas limits under and over the threshold, and that's the only error I've been getting, never a `RuntimeError::Metering` error.

Looking at [the description](https://github.com/wasmerio/wasmer/blob/80d9935aabbb0bad2f44528510732b986b520098/lib/runtime-core/src/error.rs#L178-L181) of this error, it's supposedly a bug, and the variant is only ever instantiated if a panic happens [here](https://github.com/wasmerio/wasmer/blob/80d9935aabbb0bad2f44528510732b986b520098/lib/runtime-core/src/fault.rs#L285) or when an error happens [here](https://github.com/wasmerio/wasmer/blob/80d9935aabbb0bad2f44528510732b986b520098/lib/runtime-core/src/typed_func.rs#L594).

Am I supposed to handle this error as an out-of-gas error, or is this a bug in singlepass?

**Comment by syrusakbary** [comment]
We are trying out Github discussions and we just transferred this as our first issue!
